File: app/src/main/java/com/gystry/pjetpack/ui/mine/MineFragment.java
package com.gystry.pjetpack.ui.mine;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

import com.gystry.libnavannotation.FragmentDestination;
import com.gystry.pjetpack.R;
import com.gystry.pjetpack.databinding.FragmentMineBinding;
import com.gystry.pjetpack.model.User;
import com.gystry.pjetpack.ui.home.HomeViewModel;
import com.gystry.pjetpack.ui.login.UserManager;
import com.gystry.pjetpack.utils.StatusBar;

@FragmentDestination(pageUrl = "main/tabs/my", asStarter = false)
public class MineFragment extends Fragment {

    private MineViewModel mineViewModel;
    private FragmentMineBinding mBinding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        mBinding = FragmentMineBinding.inflate(inflater, container, false);
        return mBinding.getRoot();
    }

    @SuppressLint("FragmentLiveDataObserve")
    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        User user = UserManager.getInstance().getUser();
        mBinding.setUser(user);

        UserManager.getInstance().refresh().observe(this, new Observer<User>() {
            @Override
            public void onChanged(User user) {
                if (user != null) {
                    mBinding.setUser(user);
                }
            }
        });

        mBinding.actionLogout.setOnClickListener(v -> new AlertDialog.Builder(getContext())
                .setMessage(getString(R.string.fragment_my_logout))
                .setPositiveButton(getString(R.string.fragment_my_logout_ok), (dialog, which) -> {
                    dialog.dismiss();
                    UserManager.getInstance().logout();
                    getActivity().onBackPressed();
                }).setNegativeButton(getString(R.string.fragment_my_logout_cancel), null)
                .create().show());


    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        StatusBar.lightStatusBar(getActivity(), false);
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onHiddenChanged(boolean hidden) {
        super.onHiddenChanged(hidden);
        StatusBar.lightStatusBar(getActivity(), hidden);
    }
}

File: appkt/src/main/java/com/gystry/appkt/ui/mine/MineFragment.kt
package com.gystry.appkt.ui.mine

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.gystry.appkt.R
import com.gystry.appkt.ui.dashboard.DashboardViewModel
import com.gystry.libnavannotation.FragmentDestination

/**
 * @author gystry
 * 创建日期：2021/3/29 10
 * 邮箱：gystry@163.com
 * 描述：
 */
@FragmentDestination(pageUrl = "main/tabs/my",asStarter = false)
class MineFragment : Fragment() {
    private lateinit var dashboardViewModel: DashboardViewModel

    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? {
        dashboardViewModel =
                ViewModelProviders.of(this).get(DashboardViewModel::class.java)
        val root = inflater.inflate(R.layout.fragment_dashboard, container, false)
        val textView: TextView = root.findViewById(R.id.text_dashboard)
        dashboardViewModel.text.observe(viewLifecycleOwner, Observer {
            textView.text = it
        })
        return root
    }
}